stl/cpp_on_shore/sheldon/src/task_6/main.cpp

```diff
@@ -0,0 +1,73 @@
+#include <iostream>
+#include <vector>
+
+template <typename T>
+class MyIterator {
+ public:
+  typedef T value_type;
```

Comment:
maybe use using and typename is better

Reply:
Sorry mentor, I met a bug when I modified the above code to `typedef typename T value_type;`. The bug message is `error: expected nested-name-specifier before 'T'`. 

I googled this bug and found out that this bug occured with gcc, but not MSVC. 

With gcc, this bug showed that after `typename` keyword I should use some nested name of a template parameter, like `typedef typename class::T value_type`. But in my case, it seems not necessary to use `typename`. Did I misunderstand `typename`?

Reply:
yes, in this case , we cann't use typename.